### PR TITLE
Feature/register

### PR DESCRIPTION
**Pull request by avromanov2009**

### CODE REVIEW

File: src/routes/companies.js [hunk 1]
```diff
@@ -28,15 +49,22 @@ router.post(
...
 );
 
-router.put("/:companyId", async (req, res) => {
-  const { userId } = req.body;
-  const { companyId } = req.params;
-  const company = await models.Company.addUser(companyId, userId);
-  return res.status(200).json(company);
-});
+router.put(
+  "/adduser/:companyId",
```
[the remark this answers is not in the document]
Comment (avromanov2009) [reply src/routes/companies.js]: How i can to process action in router at the feature, if i will have many that actions? Will i need add to model some function as property and run it by `model.User[action]`?

File: src/models/company.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  return await Company.findByIdAndUpdate(companyId, updatedCompany, {
+    new: true
+  });
+};
+
+companySchema.statics.searchCompanyByName = async function(name) {
+  return await Company.findOne({ name: name }).populate("users");
+};
+
+companySchema.statics.createCompany = async function(name, type, users = {}) {
```
Comment (dachia) [inline src/models/company.js]: why do you need createCompany? Company.create() does the same thing, no?

File: src/models/user.js [hunk 3]
```diff
@@ -1,12 +1,35 @@
...
+  { collection: "users" }
+);
+
+function cryptPassword(password) {
+  return bcrypt.hash(password, 12);
+}
+
+userSchema.statics.createWithPassword = async function(data) {
+  data.password = await cryptPassword(data.password);
+  try {
```
Comment (dachia) [inline src/models/user.js]: why `try; catch` ?

File: src/routes/companies.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+  async (req, res) => {
+    const company = await models.Company.create(req.body);
+    if (company instanceof TypeError)
+      throw boom.badData("Can't create company");
+    return res.status(201).json(company);
+  }
+);
+
+router.put(
+  "/:companyId",
```
Comment (dachia) [inline src/routes/companies.js]: for actions, `POST: /:companyId/actions`. `PUT: /:companyId` is to update company data

File: src/models/company.js (hunk 2 above)
Comment (avromanov2009) [reply src/models/company.js]: Will be deleted. Yes... It's not necessary.

**Comment by dachia** [comment]
Great job! 👍

**Pull request merged**